src/map/GMap.js

```diff
@@ -455,6 +649,66 @@
     GMap.prototype.zoomToFit = function () {
         return this.zoomTo(this.data());
     };
+    
+    GMap.prototype.drawingOptions = function(_) {
+        if(typeof(_) === "object") {
+            if (typeof(this._drawingManager) === "object") {
+                this._drawingManager.setOptions(_);
+            }
+        }
+        return this._drawingManager;
+    };
+    
+    GMap.prototype.userShapeSelection = function (_) {
+        if (!arguments.length) return this._userShapeSelection;
+        if (this._userShapeSelection) {
+            this._userShapeSelection.setEditable(false);
+        }
+        this._userShapeSelection = _;
+        if (this._userShapeSelection) {
+            this._userShapeSelection.setEditable(true);
+        }
+    };
+    
+    GMap.prototype.deleteUserShape = function (_) {
+        if (this._userShapeSelection === _) {
+            this.userShapeSelection(null);
+        }
+        this._userShapes.remove(_);
+    };
+    
+    GMap.prototype.onDrawingComplete = function(event) {
+        if (event.type !== google.maps.drawing.OverlayType.MARKER) {
+            this._drawingManager.setDrawingMode(null);
+            var newShape = event.overlay;
+            newShape.__hpcc_type = event.type;
+            this._userShapes.add(newShape);
+            var context = this;
+            var ctrl = false;
+            window.addEventListener('keydown', function(e) {
+                if (e.keyIdentifier === "Control" || e.ctrlKey === true) {
+                    ctrl = true;
+                }
+            });
+            window.addEventListener('keyup', function(e) {
+                if (e.ctrlKey === false) {
+                    ctrl = false;
+                }
+            });
+            google.maps.event.addListener(newShape, 'click', function (ev) {
+                context.userShapeSelection(newShape);
+                if (ev && ctrl === true) {
```

Comment:
For clarity's sake, the reason that the "ev" parameter having the ctrl key state is a problem is because it is a google maps event object, not a DOM event object.  Essentially, "ev" contains an instance of "MouseEvent" but that property is not documented, and its name may not be the same on each execution.  Furthermore there is no method exposed by the google maps event object for accessing the underlying DOM event.  The workaround that I've seen most frequently in searching for a solution is to register separate keyup and keydown handlers to manage ctrl key state.